### ISSUE TITLE
`ClickListener` not working after game lost focus while touching down

### ISSUE THREAD

**Issue by akivamu** [libgdx/libgdx#4866]
#### Issue details
Issue occurs to scene2d Button. Click event doesn't fired after losing focus to screen.

#### Reproduction steps/code
  - Create a simple app: one `Screen` with one `Stage` with two `Button` (see code)
  - Step 1: Touch down both 2 buttons at the same time (don't release yet)
  - Step 2: Press `Home` button on your phone (2 buttons still touch down)
  - Step 3: Reopen app, now 1 button can't fire click event anymore (the button that you touch last)

Below code registers click events that do change background randomly for testing. After go through these reproduce steps, one button just can't change background anymore.

```
class MyGdxGame extends ApplicationAdapter {
    Stage stage;

    @Override
    public void create() {
        stage = new Stage(new StretchViewport(800, 480));

        TextButton.TextButtonStyle style = new TextButton.TextButtonStyle();
        style.font = new BitmapFont();

        TextButton button1 = new TextButton("11111", style);
        button1.setSize(stage.getWidth() / 2, stage.getHeight());
        button1.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                Gdx.gl.glClearColor(MathUtils.random(), MathUtils.random(), MathUtils.random(), 1);
                Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
            }
        });

        TextButton button2 = new TextButton("22222", style);
        button2.setSize(stage.getWidth() / 2, stage.getHeight());
        button2.setX(stage.getWidth() / 2);
        button2.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                Gdx.gl.glClearColor(MathUtils.random(), MathUtils.random(), MathUtils.random(), 1);
                Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
            }
        });

        stage.addActor(button1);
        stage.addActor(button2);
        Gdx.input.setInputProcessor(stage);
    }

    @Override
    public void render() {
        stage.act();
        stage.draw();
    }
}
```
#### Version of LibGDX and/or relevant dependencies
LibGdx 1.9.6

#### Please select the affected platforms
- [x] Android
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [ ] MacOS

**Comment by piotr-j** [comment]
Can confirm, pressing both buttons seem to fix it.

In the meantime, you can do something like that, force touch up so buttons dont get stuck.
```java
@Override public void pause () {
    super.pause();
    for (int i = 0; i < 10; i++) {
        stage.touchUp(0, 0, i, 0);
    }
}
```

**Comment by MrStahlfelge** [comment]
Can confirm it is still a thing. For some reason, iOS is not concerned, which is surprising.
Can't check on desktop. GWT backend is not concerned because multitouch doesn't seem to be there.

**Comment by MrStahlfelge** [comment]
Relates to #5222 - the wrong behaviour for ACTION_CANCEL was fixed for polling only, event based input is still missing

**Comment by MrStahlfelge** [comment]
Actually, when ACTION_cancel is fired, information about touch state is already erased becauseof 12d0b08a464. A fix should be done near the donkeyballs.

**Comment by CivilEngineeringInformation** [comment]
I can confirm that it still exists. iOS, for some reason, is unconcerned, which is strange.
I can't check on the desktop. Because multitouch does not appear to exist, the GWT backend is unconcerned.

**Comment by MrStahlfelge** [comment]
Can you check the linked PR #6871 if it fixes the problem for you?

**Comment by crykn** [comment]
@CivilEngineeringInformation's comment is a slightly altered version of your comment [above](https://github.com/libgdx/libgdx/issues/4866#issuecomment-684780387). They did the same thing [here](https://github.com/libgdx/libgdx/issues/6890#issuecomment-1159723239) and posted a spam link over at #6212, so I guess we can just ignore it.

**Comment by MrStahlfelge** [comment]
Fixed with #6871
